src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets

```diff
@@ -0,0 +1,121 @@
+
+    Replace the files to be published with versions that have been
+    passed through the linker. Also prevent removed files from being
+    included in the generated deps.json.
+    ============================================================
+    -->
+  <Target Name="_ILLink"
+          Condition=" '$(PublishTrimmed)' == 'true' "
+          DependsOnTargets="_RunILLink">
+
+    <!-- For now, use ResolvedFileToPublish as input/output. This
+         should go away in favor of a well-defined set of runtime
+         assemblies with https://github.com/dotnet/sdk/issues/3109. -->
+    <ItemGroup>
+      <_LinkedResolvedFileToPublish Include="@(_LinkedResolvedFileToPublishCandidates)" Condition="Exists('%(Identity)')" />
+      <ResolvedFileToPublish Remove="@(_ManagedAssembliesToLink)" />
+      <ResolvedFileToPublish Include="@(_LinkedResolvedFileToPublish)" />
+    </ItemGroup>
+
+    <!-- Remove assemblies from inputs to GenerateDepsFile. See
+         https://github.com/dotnet/sdk/pull/3086. -->
+    <ItemGroup>
+      <_RemovedManagedAssemblies Include="@(_ManagedAssembliesToLink)" Condition="!Exists('$(IntermediateLinkDir)/%(Filename)%(Extension)')" />
+
+      <ResolvedCompileFileDefinitions Remove="@(_RemovedManagedAssemblies)" />
+      <NativeCopyLocalItems Remove="@(_RemovedManagedAssemblies)" />
+      <ResourceCopyLocalItems Remove="@(_RemovedManagedAssemblies)" />
+      <RuntimeCopyLocalItems Remove="@(_RemovedManagedAssemblies)" />
+      <RuntimeTargetsCopyLocalItems Remove="@(_RemovedManagedAssemblies)" />
+      <RuntimePackAsset Remove="@(_RemovedManagedAssemblies)" />
+
+      <!-- Also remove them from the old deps file generation logic,
+           until https://github.com/dotnet/sdk/issues/3098 is
+           fixed. -->
+      <_PublishConflictPackageFiles Include="@(_RemovedManagedAssemblies)" />
+    </ItemGroup>
+
+
+  </Target>
+
+
+  <!--
+    ============================================================
+                     _RunILLink
+
+    Execute the linker. This target runs incrementally, only executing
+    if the output semaphore file is out of date with respect to the inputs.
+    ============================================================
+    -->
+   <UsingTask TaskName="ILLink" AssemblyFile="$(ILLinkTasksAssembly)" />
+   <Target Name="_RunILLink"
+           DependsOnTargets="_ComputeManagedAssembliesToLink"
+           Inputs="$(MSBuildAllProjects);@(_ManagedAssembliesToLink);@(TrimmerRootDescriptors);@(ReferencePath)"
+           Outputs="$(_LinkSemaphore)">
+
+     <Delete Files="@(_LinkedResolvedFileToPublishCandidates)" />
+     <ILLink AssemblyPaths="@(_ManagedAssembliesToLink)"
+             ReferenceAssemblyPaths="@(ReferencePath)"
+             RootAssemblyNames="@(IntermediateAssembly->'%(Filename)')"
+             RootDescriptorFiles="@(TrimmerRootDescriptors)"
+             OutputDirectory="$(IntermediateLinkDir)"
+             ExtraArgs="$(ExtraLinkerArgs) --skip-unresolved true" />
```

Comment:
Should this be ExtraTrimmerArgs if the other public facing options are "Trimmer"?

Reply:
Named it `_ExtraTrimmerArgs`. I don't want to advertise this as a public facing option, and am hoping we can stop using it in the future in favor of defaults that do the right thing. Taking a property here makes it easier to test and makes it easier for us to give users workarounds in case they encounter issues.